make_site.py

```diff
@@ -236,39 +249,51 @@ class InfoEntry:
 num_thms = len([d for d in declarations if declarations[d].info.kind == 'theorem'])
 num_defns = len(declarations) - num_thms
 
-if DOWNLOAD:
-    download(
-        'https://leanprover-community.github.io/mathlib4_docs/100.yaml',
-        DATA/'100.yaml')
-    with (DATA/'100.yaml').open('r', encoding='utf-8') as h_file:
-        hundred_theorems = [HundredTheorem(thm,**content) for (thm,content) in yaml.safe_load(h_file).items()]
-        for h in hundred_theorems:
-            if h.decl:
-                assert not h.decls
-                h.decls = [h.decl]
-            if h.decls:
-                doc_decls = []
-                for decl in h.decls:
-                    try:
-                        decl_info = declarations[decl]
-                    except KeyError:
-                        print(f'Error: 100 theorems entry {h.number} refers to a nonexistent declaration {decl}')
-                        continue
-                    # note: the `.bmp` data files use doc-relative links
-                    header = decl_info.header.replace('href="./Mathlib/', 'href="./mathlib4_docs/Mathlib/')
-                    doc_decls.append(DocDecl(
-                        name=decl,
-                        decl_header_html = header,
-                        # note: the `.bmp` data files use doc-relative links
-                        docs_link='/mathlib4_docs/' + decl_info.info.docLink,
-                        src_link=decl_info.info.sourceLink))
-                h.doc_decls = doc_decls
-            else:
-                h.doc_decls = []
-    pkl_dump('hundred_theorems', hundred_theorems)
-else:
-     hundred_theorems = pkl_load('hundred_theorems', dict())
+class NTheorems(Enum):
+    Hundred = auto()
+    ThousandPlus = auto()
 
+def download_N_theorems(kind: NTheorems) -> dict:
+    if kind == NTheorems.Hundred:
+        (fname, Type, name) = ('100.yml', HundredTheorem, 'hundred_theorems')
+    else:
+        (fname, Type, name) = ('1000.yml', ThousandPlusTheorem, 'thousand_theorems')
```

Comment:
Both https://leanprover-community.github.io/mathlib4_docs/1000.yml or https://leanprover-community.github.io/mathlib4_docs/1000.yaml are 404 errors - is there a PR to doc-gen this depends on? (Sorry if I missed it somewhere above)

Reply:
@grunweg It seems easier to just read the data from https://raw.githubusercontent.com/leanprover-community/mathlib4/refs/heads/master/docs/100.yaml and https://raw.githubusercontent.com/leanprover-community/mathlib4/refs/heads/master/docs/1000.yaml ?

Reply:
@bryangingechen Good point. I checked; this needs a PR to doc-gen, here it is (almost trivial diff):
https://github.com/leanprover-community/mathlib4_docs/pull/5

Reply:
@fpvandoorn Good question. I'm not sure which way is best. I like the conceptual separation of "docs builds mathlib and makes those overview files available, the webpage only reads the docs and not the mathlib repo". Conceptually, these overview files feel similar to generated documentation to me (both are externally useful artifacts, derived from the mathlib source in some sense --- one could imagine generating 100.yaml automatically from a `100theorems` attribute, for instance).
In other words: the current system makes some sense to me, and I'm not sure if there's a *good* reason to change it. "You need to just through another hoop" could be one; perhaps this just needs to be documented better?

In any case, I'd want the handling for `{undergrad,overview}.yaml` to be consistent with this --- so I'd want to change two more places in the code. That's not a big issue though.

Reply:
I'm happy to discuss which of these locations makes the most sense. But perhaps that should be a broader discussion/I'd like to have at least one more opinion here.

Reply:
Here's the change where we first started reading data files from mathlib_docs: https://github.com/leanprover-community/leanprover-community.github.io/pull/125

And [here's the relevant Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/website.20overview/near/208659351), where @robertylewis proposes the idea of hosting the yaml files with doc-gen. I'm guessing the idea of linking directly to mathlib files at raw.githubusercontent.com just didn't occur to us at the time. Using the raw.githubusercontent.com links has the advantage of making things more direct, although as you say, maybe someone external is already depending on the files at mathlib4_docs.

Reply:
Ok, that's fine with me as well.

I'm happy with this PR if Bryan is.

Reply:
Yes, I'm happy with this as is now!

Reply:
Thanks for digging up the context. I've read the zulip thread and the PRs now. If I understand correctly, there is a synchronisation constraint: it would be nice to keep the declarations (generated from doc-gen) and the web-page in sync --- so retrieving both from doc-gen4 seems reasonable.
I don't know how often doc-gen fails these days - but if it did, having a webpage using the old data consistently (rather than some weird mixture, and having to think about all corner cases this can create) strikes me as a better trade-off. If failing docs are an issue, setting up an alarm on failures seems better.

TL;DR: I think the current design is better than the other version, and will stick to it.